### PR TITLE
feat(config): js catcher options support

### PR DESCRIPTION
**Pull request by neSpecc**
You can pass `user`, `context`, `beforeSend` and other [JS Catcher options](https://github.com/codex-team/hawk.javascript?tab=readme-ov-file#usage) via `catcherOptions` config property.

```ts
export default defineNuxtConfig({
  modules: [
    '@hawk.so/nuxt'
  ],

  hawk: {
    token: process.env.HAWK_TOKEN,
    catcherOptions: {
      context: {
        // any data you want to send with all events
        appName: 'Hawk Nuxt Playgorund',
      },
      // method for filtering/modifying an event to be sent
      beforeSend: (event) => {
        event.context.appVersion = '1.0.0'

        return event
      },
    },
  },
})
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +35,34 @@ export default defineNuxtConfig({
...
+  modules: [
+    '@hawk.so/nuxt'
+  ],
+
+  hawk: {
+    token: process.env.HAWK_TOKEN,
+    catcherOptions: {
+      context: {
+        // any data you want to send with all events
+        appName: 'Hawk Nuxt Playgorund',
```
Comment (TatianaFomina) [inline README.md]:
```suggestion
        appName: 'Hawk Nuxt Playground',
```

File: README.md [hunk 2]
```diff
@@ -35,6 +35,34 @@ export default defineNuxtConfig({
...
+      },
+      // method for filtering/modifying a sending event
```
Comment (TatianaFomina) [inline README.md]:
```suggestion
      // method for filtering/modifying a sent event
```

File: playground/nuxt.config.ts [hunk 3]
```diff
@@ -1,7 +1,20 @@
 export default defineNuxtConfig({
   modules: ['../src/module'],
+
   hawk: {
     token: process.env.HAWK_TOKEN,
+    catcherOptions: {
+      context: {
+        appName: 'Hawk Nuxt Playgorund',
```
Comment (TatianaFomina) [inline playground/nuxt.config.ts]:
```suggestion
        appName: 'Hawk Nuxt Playground',
```

**Review by TatianaFomina — APPROVED** [review]

File: src/runtime/plugin.client.ts [hunk 4]
```diff
@@ -30,11 +31,13 @@ export default defineNuxtPlugin((nuxtApp) => {
...
 
-  const hawkInstance = new HawkCatcher({
+  const hawkInstance = new HawkCatcher(Object.assign({}, hawkConfig.catcherOptions, {
     token: hawkConfig.token,
     vue: nuxtApp.vueApp,
     release: getReleaseId(),
-  })
+  }, {
+    beforeSend,
+  }))
```
Comment (e11sy) [inline src/runtime/plugin.client.ts:40]: is object assign safe for `hawkConfig.catcherOptions`, as i know this is not deep cloning, maybe we should assert `HawkCatcher`  separately?

Comment (neSpecc) [reply src/runtime/plugin.client.ts:40]: we don't need deep merging there, just overriding values

Comment (neSpecc) [reply src/runtime/plugin.client.ts:40]: and `catcherOptions` are assigning to the empty object, so there will be no overrides at all

**Pull request merged**